crates/maple_cli/src/cmd/rpc/filer.rs

```diff
@@ -1,67 +1,102 @@
-use std::{fs, io};
-
+use super::{write_response, Message};
 use anyhow::Result;
+use icon::prepend_filer_icon;
+use serde::{Deserialize, Serialize};
 use serde_json::json;
+use std::path::PathBuf;
+use std::{fs, io};
 
-use super::{write_response, Message};
-use icon::prepend_filer_icon;
+/// Display the inner path in a nicer way.
+struct DisplayPath {
+    inner: PathBuf,
+    enable_icon: bool,
+}
 
-fn into_string(entry: std::fs::DirEntry, enable_icon: bool) -> String {
-    let path = entry.path();
-    let path_str = if path.is_dir() {
-        format!(
-            "{}/",
-            path.file_name().and_then(std::ffi::OsStr::to_str).unwrap()
-        )
-    } else {
-        path.file_name()
-            .and_then(std::ffi::OsStr::to_str)
-            .map(Into::into)
-            .unwrap()
-    };
+impl DisplayPath {
+    pub fn new(path: PathBuf, enable_icon: bool) -> Self {
+        Self {
+            inner: path,
+            enable_icon,
+        }
+    }
+
+    #[inline]
+    fn to_file_name_str(&self) -> Option<&str> {
+        self.inner.file_name().and_then(std::ffi::OsStr::to_str)
+    }
+}
+
+impl Into<String> for DisplayPath {
+    fn into(self) -> String {
+        let path_str = if self.inner.is_dir() {
+            // TODO: windows compatible?
+            format!("{}/", self.to_file_name_str().unwrap())
```

Comment:
`format!("{}{}", self.to_file_name_str().unwrap(), std::path::MAIN_SEPARATOR)`

Reply:
Thanks for the comment. I also picked the nits from #423 into this PR to reduce the changes in that PR.